### PR TITLE
Hook in Automattic Tracks for analytics in internal beta

### LINKED ISSUE #3511: Integrate Automattic Tracks as a WPAnalytics Tracker
Pull in the Automattic-Tracks-iOS client and implement a WPAnalytics tracker with it.

### PR DESCRIPTION
**Pull request by astralbodies**
Closes #3511 

This integrates the first beta release of the Nosara Tracks client with the WPAnalytics tracker setup. This will only work with DEBUG and INTERNAL_BUILD builds. I need to start tracking real data so putting it into the internal build would be very useful.

### CODE REVIEW

File: WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+}
+
+- (void)endSession
+{
+    
+}
+
+- (void)refreshMetadata
+{
+    NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
```
Comment (sendhil) [inline WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m]: Perhaps we could reduce the duplication with the code over https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerMixpanel.m#L84? Maybe some kind of class to generate this dictionary for us?

**Comment by sendhil** [comment]
Aside from my comment looks good. Tests would be nice though ;)

**Comment by astralbodies** [comment]
@sendhil I added a conditional macro to enable Tracks only on Debug & Release_Internal builds. Also the Podfile was updated to only include the code in those build configurations.

I could add a unit test suite for the Tracker but I wasn't exactly sure what I would be testing at that point :smile:. 

I'll improve the Trackers with the issue regarding Mixpanel firing off the app install event since its more of an architecture issue not integration with Tracks.

**Comment by sendhil** [comment]
@astralbodies  sounds good - we'll merge this in for now and you can come in and patch that part later.

**Pull request merged**